src/utils/actions.js

```diff
@@ -0,0 +1,396 @@
+      const resourceType = input.type;
+      // assume subjectReference might constrain the search
+      const reference = input.subjectReference?.reference;
+      let uri = resourceType;
+      if (reference) {
+        uri += `?subject=${reference}`;
+      }
+      const resources = client.read(uri);
+      const dateFilter = input.dateFilter;
+      const filteredResources = [];
+      // TODO: most of this filtering should be done
+      // through fhirPath when the resource query
+      // is made.
+
+      if (dateFilter) {
+        // assume two filters don't overlap
+        const path = dateFilter.path;
+        const searchParam = dateFilter.searchParam;
+        const filterPeriod = dateFilter.valuePeriod; // assume its a valuePeriod
+        if (path) {
+          const dateFiltered = resources.filter(resource => {
+            // get value by path
+            const date = getByPath(resource, path); // assume its a datetime?
+            return compareDates(date, filterPeriod);
+          });
+          filteredResources.push.apply(filteredResources, dateFiltered);
+        } else {
+          // path or searchParam are required
+          const dateFiltered = resources.filter(resource => {
+            // get value by param
+            const date = getBySearchParam(resource, searchParam); // assume its a datetime?
+            return compareDates(date, filterPeriod);
+          });
+          filteredResources.push.apply(filteredResources, dateFiltered);
+        }
+      } else {
+        // if there's no date filter, we can just push all the resources
+        // in to be filtered by code
+        filteredResources.push.apply(filteredResources, resources);
+      }
+
+      const codeFilter = input.codeFilter;
+      if (codeFilter) {
+        // code filter isn't under the same constraint as datefilter
+        // we can assume that it's just a path
+        const path = codefilter.path;
+        const searchParam = codeFilter.searchParam;
+        if (path) {
+          // no need to refilter resources
+          const filteredResources = filteredResources.filter(resource => {
+            const code = getByPath(resource, path);
+            return compareCodes(code, codeFilter.code); // codeFilter.code is of type Coding, could also be a valueSet?
+          });
+        } else if (searchParam) {
+          const filteredResources = filteredResources.filter(resource => {
+            const code = getBySearchParam(resource, searchParam);
+            return compareCodes(code, codeFilter.code);
+          });
+        }
+      }
+      return context.records.push(filteredResources);
+    });
+  }
+}
+function executeResearchQuery(context) {
+  // TODO: what's a datamart?
+}
+function getByPath(resource, path) {
+  // TODO: function to search resources by path
+  // and return correct attribute
+  return resource;
+}
+
+function compareCodes(code1, code2) {
+  // assume they're both coding
```

Comment:
Might as well name it as such
```suggestion
function compareCodes(coding1, coding2) {
```